src/System.Runtime.InteropServices.RuntimeInformation/src/ArchitectureManager.Unix.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.ComponentModel;
+
+namespace System.Runtime.InteropServices
+{
+    internal partial struct ArchitectureManager
+    {
+        private static Architecture ProcessArchCore
+        {
+            get
+            {
+                if (IsARM)
+                {
+                    return Architecture.ARM32;
+                }
+                else
+                {
+                    return _is32BitProcess ? Architecture.BIT32 : Architecture.BIT64;
+                }
+            }
+        }
+
+        private static Architecture OSArchCore
+        {
+            get
+            {
+                string arch = null;
+                int result = Interop.Sys.UName(out arch);
+                if (result >= 0 && arch != null)
+                {
+                    if (arch.IndexOf("arm", StringComparison.OrdinalIgnoreCase) >= 0)
+                        return Architecture.ARM32;
+
+                    if (arch.IndexOf("x86_64", StringComparison.OrdinalIgnoreCase) >= 0)
+                        return Architecture.BIT64;
+
+                    return Architecture.BIT32;
+                }
+                else
+                {
+                    // uname failed, returned -1 or arch is not set.
+                    throw new Win32Exception();
+                }
+            }
+        }
+
+        private static bool IsARM
```

Comment:
I don't understand IsARM.  Why is ARM special?  While IA-64 isn't likely to be something we port to, maybe someone wants to port to SPARC (more likely SPARC64), or some new exciting thing comes along next year; etc.  There's already a version that returns an enum, why isn't that sufficient?